x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/SecurityRestFilter.java

```diff
@@ -76,19 +72,12 @@ public void handleRequest(RestRequest request, RestChannel channel, NodeClient c
         }
 
         final RestRequest wrappedRequest = maybeWrapRestRequest(request);
-        authenticationService.authenticate(wrappedRequest.getHttpRequest(), ActionListener.wrap(authentication -> {
-            if (authentication == null) {
-                logger.trace("No authentication available for REST request [{}]", request.uri());
-            } else {
-                logger.trace("Authenticated REST request [{}] as {}", request.uri(), authentication);
+        auditTrailService.get().authenticationSuccess(wrappedRequest);
+        secondaryAuthenticator.authenticateAndAttachToContext(wrappedRequest, ActionListener.wrap(secondaryAuthentication -> {
```

Comment:
I don't think it technically matters, but would it make more sense to keep secondary auth happen at the same time as primary auth ?  (can also be a follow up)

Reply:
I think technically it's possible but not necessary, so, in order to minimize the scope of this change I have not implemented. I propose we consider this optional and discuss when all requiered pieces are merged (I'll take it to track these optionals somehow).